From 4e7ebe21bb15cb27ffdf582196dec20732fdd242 Mon Sep 17 00:00:00 2001
From: Joshua Ashton <joshua@froggi.es>
Date: Thu, 2 Nov 2023 07:32:16 +0000
Subject: [PATCH 39/41] drm/amd/display: Remove erroneous color_mgmt_changed
 check in amdgpu_dm_plane_set_color_properties

Pure luck we didn't run into this before.

Could be a newly allocated dc_plane_state.

Signed-off-by: Joshua Ashton <joshua@froggi.es>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c
index c84692612..86a2d449c 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c
@@ -1031,8 +1031,15 @@ amdgpu_dm_plane_set_color_properties(struct drm_plane_state *plane_state,
 	int ret;
 
 	/* We have nothing to do here, return */
+	/*
+	 * JoshA: WE HAVE TO DO THIS EVERY TIME.
+	 * It's on a new dc_plane_state allocation, none of this data is here!
+	 * !!!!!!!!
+	 * This was always true before we duped properties due to LUCK and the
+	 * properties matching.
 	if (!plane_state->color_mgmt_changed)
 		return 0;
+	*/
 
 	dc_plane_state->hdr_mult = dc_fixpt_from_s3132(dm_plane_state->hdr_mult);
 
-- 
2.43.0.232.ge79552d197

